### DISCUSSION TITLE
Different implementation of Pipeline

### DISCUSSION THREAD

**Discussion by michael-rubel** [chefhasteeth/pipeline#1] [Show and tell]
Hi @chefhasteeth 👋

I've used parts of your code to implement a slightly [different version](https://github.com/michael-rubel/laravel-enhanced-pipeline) of pipeline based on the original one available in the framework, with an additional `onFailure` method. Added you to credits of course. 🙂

Thank you for open-sourcing your work!

**Comment by chefhasteeth** [comment]
Nice. I had considered adding callbacks for success/failure (as well as introducing a `PipelineStatus` enum and `passed()` / `failed()` methods, etc) but in the end decided to go with a "vanilla" approach. Glad you found it useful. :+1:

**Reply by michael-rubel** [reply]
My idea was also to fire events throughout the pipeline lifecycle. It would probably be a nice addition for event-driven apps. We'll see if it fits reality. 🙂
